### PR TITLE
Feature/ak/batch job filelist

### PR DESCRIPTION
**Pull request by Akshith-Patil**
## Proposed changes

Created new file management design for upload section
1. Added 'search' bar for upload section
2. New 'remove all' button to delete all uploaded files
3. Thumbnails for image and video files
4. Pagination for Batch Job upload section
5. Modal popup to view files in full-sized view

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] DevOps related

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [project guidelines](https://github.com/Visual-Testing-for-Android-Apps/Project-guidelines) doc
- [x] I locally linted my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

**Before Uploading Files**
Before Upload

**After Uploading files**
After Upload

After clicking on the file thumbnail
Modal Popup

### CODE REVIEW

File: src/components/reportPage/upload.css [hunk 1]
```diff
@@ -120,4 +283,61 @@
...
+    padding: 0%;
+  }
+
+  .file-name {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    font-size: 0.875rem;
+  }
+
+  .modalContent {
```
Comment (mt-empty) [inline src/components/reportPage/upload.css:312]: should not be Camel case

**Comment by Akshith-Patil** [comment]
> This fails to build, can you investigate?



> This fails to build, can you investigate?
There was an issue with the paginate package, but it has been fixed now

File: src/components/reportPage/UploadBox.jsx [hunk 2]
```diff
@@ -2,138 +2,44 @@ import React, { useCallback, useEffect, useState } from "react";
...
-    }
-
     // Returns null if files should be accepeted
     return null;
   };
-
   const acceptedFileTypes = ["image/jpeg", "image/png", "video/mp4"];
-  const maxFiles = fileLimit ? fileLimit : 0; // 0 unlimited files
-  const maxSize = maxFileSize ? maxFileSize * 1000 * 1000 : Infinity; // 0 is unlimited file size, in bin bytes
+  const maxFiles = props.fileLimit == Infinity ? 0 : props.fileLimit;
```
Comment (mt-empty) [inline src/components/reportPage/UploadBox.jsx:38]: This is Ok, but I would rather have an upper limit, having `Infinity` might break the website in ways we do not anticipate.

**Review by idomo1 — CHANGES_REQUESTED** [review]
Since you reverted merging in develop, if this PR is merged it will undo a bunch of work (have a look at the diff).
You might need to revert the revert commit

**Pull request closed without merging**